UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/Controllers/BaseController.cs

```diff
@@ -33,6 +36,58 @@ protected IActionResult BuildBadRequestErrorResponse(List<Error> errors)
                 CorrelationId = GetCurrentCorrelationId()
             });
         }
+        public class InternalServerErrorObjectResult : ObjectResult
+        {
+            public InternalServerErrorObjectResult(object value) : base(value)
+            {
+                StatusCode = StatusCodes.Status500InternalServerError;
+            }
+        }
+
+        protected IActionResult BuildInternalServerErrorResponse()
+        {           
+            return new InternalServerErrorObjectResult(new InternalServerError
+            {
+               CorrelationId = GetCurrentCorrelationId(),
+               Detail = InternalServerError,
+            });
+        }
+
+        public class NotModifiedObjectResult : ObjectResult
+        {
+            public NotModifiedObjectResult(object value) : base(value)
+            {
+                StatusCode = StatusCodes.Status304NotModified;
+            }
+        }
+
+        protected IActionResult BuildNotModifiedResponse()
+        {            
+            return new NotModifiedObjectResult(new ErrorDescription {              
+            });
+        }
+
+        protected IActionResult GetEssResponse(ExchangeSetServiceResponse model, List<Error> errors = null)
+        {
+            HttpStatusCode code = model.HttpstatusCode;           
+            switch (code)
+            {
+                case (HttpStatusCode)(int)HttpStatusCode.OK:
+                    if (model.LastModified != null)
+                        httpContextAccessor.HttpContext.Response.Headers.Add(LastModifiedDateHeaderKey, model.LastModified);
+                    return Ok(model.ExchangeSetResponse);
+                case (HttpStatusCode)(int)HttpStatusCode.InternalServerError:
+                    return BuildInternalServerErrorResponse();                    
+                case (HttpStatusCode)(int)HttpStatusCode.BadRequest:
+                    return BuildBadRequestErrorResponse(errors);
+                case (HttpStatusCode)(int)HttpStatusCode.NotModified:
+                    if (model.LastModified != null)
+                        httpContextAccessor.HttpContext.Response.Headers.Add(LastModifiedDateHeaderKey, model.LastModified);
+                    return BuildNotModifiedResponse();
+                default:
+                   return BuildInternalServerErrorResponse();
```

Comment:
This block of code might be easier to read with a line break after each return statement?

Reply:
Modified code for line break after each return statement.